src/core/Carousel/Carousel.tsx

```diff
@@ -0,0 +1,494 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import cx from 'classnames';
+import {
+  getRandomValue,
+  useIntersection,
+  useMergedRefs,
+  useTheme,
+} from '../utils';
+import { IconButton } from '../Buttons';
+import SvgChevronLeft from '@itwin/itwinui-icons-react/cjs/icons/ChevronLeft';
+import SvgChevronRight from '@itwin/itwinui-icons-react/cjs/icons/ChevronRight';
+import '@itwin/itwinui-css/css/carousel.css';
+
+export type CarouselProps = {
+  /**
+   * Index of the currently shown slide.
+   * Can be used to set the default index or control the active slide programmatically.
+   * @default 0
+   */
+  activeSlideIndex?: number;
```

Comment:
Then we also need `onSlideChange`

Reply:
Is it worth it? `onSlideChange` is present in the dots

Reply:
For me it looks odd to set index in one place and have callback in other.

Reply:
Added